packages/machine/src/instruction-executor.ts

```diff
@@ -31,15 +31,15 @@ export class InstructionExecutor implements Observable {
    * The underlying state for the entire machine. All state here is a result of
    * a completed and commited protocol.
    */
-  public nodeState: NodeState;
+  public node: Node;
 
   // Observable
   public observers: Map<NotificationType, Function[]> = new Map();
 
   constructor(config: InstructionExecutorConfig) {
     this.responseHandler = config.responseHandler;
-    this.nodeState = new NodeState(config.state || {}, config.network);
-    this.middleware = new Middleware(this.nodeState, config.opGenerator);
+    this.node = new Node(config.state || {}, config.network);
```

Comment:
This seems like a bizarre decision, why would this class have a Node?

Reply:
Handle to the context it's executing in.

Reply:
Can't we pass in only the minimum amount of information it needs? Like what does this class actually require to know?

Reply:
We can, hence the WIP :)